src/preset_cli/cli/superset/sync/dbt/metrics.py

```diff
@@ -399,3 +393,32 @@ def get_models_from_sql(
             raise ValueError(f"Unable to find model for SQL source {table}")
 
     return [model_map[ModelKey(table.db, table.name)] for table in sources]
+
+
+def remove_line_break_wrappers(sql: str) -> str:
+    """
+    Remove leading and trailing line breaks from SQL statements.
+    """
+    pattern = r"^\s*\n|\n\s*$"
+    return re.sub(pattern, "", sql, flags=re.MULTILINE)
+
+
+def replace_metric_syntax(
+    sql: str,
+    dependencies: List[str],
+    metrics: Dict[str, MetricSchema],
+) -> str:
+    """
+    Replace metric keys with their SQL syntax.
+    This method is a fallback in case ``sqlglot`` raises a ``ParseError``.
+    """
+    for parent_metric in dependencies:
+        parent_metric_name = parent_metric.split(".")[-1]
+        pattern = r"\b" + re.escape(parent_metric_name) + r"\b"
+        parent_metric_syntax = get_metric_expression(
+            parent_metric_name,
+            metrics,
+        )
+        sql = re.sub(pattern, parent_metric_syntax, sql)
```

Comment:
Same syntax I previously had inside `get_metric_expression` (just moved to its own method).